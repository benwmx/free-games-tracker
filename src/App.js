import React from 'react';
import { HashRouter as Router, Route, Switch } from 'react-router-dom';
import Header from './components/Header';
import Games from './components/Games';
import Details from './components/Details';
import About from './components/About';
import ScrollButton from './components/ScrollButton';

function App() {
  return (
    <>
      <Router>
        <Header />
        <Switch>
          <Route path="/" exact>
            <Games />
          </Route>
          <Route path="/details">
            <Details />
          </Route>
          <Route path="/about">
            <About />
          </Route>
        </Switch>
        <ScrollButton />
      </Router>
    </>
  );
}

export default App;
