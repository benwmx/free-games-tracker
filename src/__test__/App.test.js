import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { HashRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import App from '../App';

const APP = (
  <Provider store={store}>
    <Router>
      <App />
    </Router>
  </Provider>
);

describe('App Component test', async () => {
  it('Should render', () => {
    const app = renderer.create(APP).toJSON();
    expect(app).toMatchSnapshot();
  });

  it('UI components', () => {
    render(APP);
    screen.queryAllByRole('link').forEach((role) => expect(role).toBeInTheDocument());
    screen.debug();
  });
});
