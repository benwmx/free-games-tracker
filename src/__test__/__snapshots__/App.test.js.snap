// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component test Should render 1`] = `
Array [
  <header
    className="header"
    style={
      Object {
        "backgroundColor": "rgba(219, 219, 219, 0.8)",
      }
    }
  >
    <a
      aria-current="page"
      className="active"
      href="#/"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        className="headerBtnsLight"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M20 20a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1v-9H1l10.327-9.388a1 1 0 0 1 1.346 0L23 11h-3v9z"
          />
        </g>
      </svg>
    </a>
    <div
      className="infoAndTheme"
    >
      <svg
        className="headerBtnsLight"
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"
        />
      </svg>
      <a
        aria-current={null}
        href="#/about"
        onClick={[Function]}
      >
        <svg
          className="headerBtnsLight"
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16zm.93-9.412-1 4.705c-.07.34.029.533.304.533.194 0 .487-.07.686-.246l-.088.416c-.287.346-.92.598-1.465.598-.703 0-1.002-.422-.808-1.319l.738-3.468c.064-.293.006-.399-.287-.47l-.451-.081.082-.381 2.29-.287zM8 5.5a1 1 0 1 1 0-2 1 1 0 0 1 0 2z"
          />
        </svg>
      </a>
    </div>
  </header>,
  <div
    className="containerLight"
  >
    <div
      className="introSearchLight"
    >
      <div
        className="intro"
      >
        <h1
          className="title"
        >
          Free Games Tracker
        </h1>
        <p>
          Games from 1997 up to now
        </p>
        <p>
          You can search by Title,Realease Date,Genre,Developer
        </p>
      </div>
      <form
        className="search"
      >
        <input
          className="searchBar"
          onChange={[Function]}
          placeholder="Search A Game"
          type="text"
        />
      </form>
    </div>
    <div
      className="gamesContainer"
    />
  </div>,
  <button
    className="scrollButton"
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="button"
  >
    <svg
      className="scrollIconLight"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 8A8 8 0 1 0 0 8a8 8 0 0 0 16 0zm-7.5 3.5a.5.5 0 0 1-1 0V5.707L5.354 7.854a.5.5 0 1 1-.708-.708l3-3a.5.5 0 0 1 .708 0l3 3a.5.5 0 0 1-.708.708L8.5 5.707V11.5z"
      />
    </svg>
  </button>,
]
`;
